### PR TITLE
eslint-plugin: Add new "graphql-require-description" rule

### PR DESCRIPTION
**Pull request by jaroslav-kubicek**
related to https://github.com/adeira/universe/issues/210

### CODE REVIEW

File: src/eslint-plugin-adeira/src/rules/__tests__/graphql-require-description.test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+          name: 'TripSegment',
+          fields: {
+            departure: {
+              type: GraphQLRouteStop,
+              resolve: ({ departure }: TripData): DepartureArrival => departure,
+            },
+          },
+        });
+  `,
+      errors: [
```
Comment (tbergq) [inline src/eslint-plugin-adeira/src/rules/__tests__/graphql-require-description.test.js]: What if I add `description: ''`

**Comment by tbergq** [comment]
Not sure how I feel about this rule, since I might add empty description, or just `adfa`. 

We cannot really validate that the description is meaningful by eslint alone. And I also believe some types are self descriptive, though I agree best practise is to have a description.

**Comment by mrtnzlml** [comment]
> Not sure how I feel about this rule, since I might add empty description, or just `adfa`.
> 
> We cannot really validate that the description is meaningful by eslint alone. And I also believe some types are self descriptive, though I agree best practise is to have a description.

Facebook uses something like `GraphQLSelfDescriptive` type for descriptions which are self-descriptive (which effectively makes every description required, not only for objects but even fields). See:

Screenshot 2020-01-15 at 13 21 44

**Comment by tbergq** [comment]
Ok, that is also good. I am not saying I am against the rule, just that, you will still need to address this on CR. Because I might just add `desicription: 'a'` just to shut up the eslint rule.

File: src/eslint-plugin-adeira/src/rules/__tests__/graphql-require-description.test.js (hunk 1 above)
Comment (jaroslav-kubicek) [reply src/eslint-plugin-adeira/src/rules/__tests__/graphql-require-description.test.js]: you are right, I realized it later, will fix, thanks!

**Comment by jaroslav-kubicek** [comment]
yes, the description doesn't add much value for objects like `Address`, `Price` etc. and I'm fine if people just put there `User` into description for `User`. My goal is to just make people think what such object represents, improve docs and be consistent.

Graphiql even puts there `No Description` as a little complain 🙂 

Screenshot 2020-01-15 13 39 53

**Comment by tbergq** [comment]
> My goal is to just make people think what such object represents, improve docs and be consistent

Yes, good things, so let's continue and merge it when the empty string is handled 😊

File: src/eslint-plugin-adeira/src/rules/__tests__/graphql-require-description.test.js (hunk 1 above)
Comment (jaroslav-kubicek) [reply src/eslint-plugin-adeira/src/rules/__tests__/graphql-require-description.test.js]: should be fixed now!

**Pull request merged**